cmds/rm/rm_test.go

```diff
@@ -38,58 +39,158 @@ var tests = []makeit{
+		return "", err
+	}
+
+	/* Create file, symlink, and block device in a new directory */
+	newD, err := ioutil.TempDir(d, tests2[2].n)
+	if err != nil {
+		return "", err
+	}
+	if err := ioutil.WriteFile(path.Join(newD, tests2[0].n), []byte(""), tests2[0].m); err != nil {
+		return "", err
+	}
+
+	if err := os.Symlink(path.Join(d, tests2[0].n), path.Join(newD, "symlink")); err != nil {
+		return "", err
+	}
+	/* Does not work on Travis
+	if err := syscall.Mknod(path.Join(newD, tests2[3].n), 0777, 64); err != nil {
+		return "", err
+	}
+	*/
 	return d, nil
 }
 
-// without any flag
+func printFiles(d string) (filenames []string, err error) {
+	var nameArray []string
+	fileNames, err := ioutil.ReadDir(d)
+	if err != nil {
+		return nil, err
+	}
+	for _, file := range fileNames {
+		fmt.Println(file.Name())
+		nameArray = append(nameArray, path.Join(d, file.Name()))
+	}
+	return nameArray, nil
+}
+
+// TEST 1
+// Flags: -v; only delete files from tests
+
 func Test_rm_1(t *testing.T) {
+	fmt.Println("TEST 1:")
 	d, err := setup()
 	if err != nil {
-		t.Fatal("Error on setup of the test: creating files and folders.")
+		t.Fatalf("Error on setup of the test: creating files and folders: %s", err)
 	}
-	defer os.RemoveAll(d)
-
 	fmt.Println("== Deleting files and empty folders (no args) ...")
 	files := []string{path.Join(d, "hi1.txt"), path.Join(d, "hi2.txt"), path.Join(d, "go.txt")}
-
-	flags.v = true
-	if err := rm(files); err != nil {
+	var flags rmFlags
+	flags.verbose = true
+	if err := rm(files, flags); err != nil {
 		t.Error(err)
 	}
+	os.RemoveAll(d)
 }
 
-// using r flag
+// TEST 2
+// Flags: -v -r; only delete files from test1
 func Test_rm_2(t *testing.T) {
+	fmt.Println("TEST 2:")
 	d, err := setup()
 	if err != nil {
-		t.Fatal("Error on setup of the test: creating files and folders.")
+		t.Fatalf("Error on setup of the test: creating files and folders.")
 	}
-	defer os.RemoveAll(d)
-
-	flags.v = true
-	flags.r = true
+	var flags rmFlags
+	flags.verbose = true
+	flags.recursive = true
 	fmt.Println("== Deleting folders recursively (using -r flag) ...")
 	files := []string{d}
-	if err := rm(files); err != nil {
+	if err := rm(files, flags); err != nil {
+		t.Error(err)
+	}
+	defer os.RemoveAll(d)
+}
+
+// TEST 3
+// Flags: -r none; delete files from test1 and 2 and print all files in directory
+func Test_rm_3(t *testing.T) {
+	fmt.Println("TEST 3:")
+	d, err := setup()
+	if err != nil {
+		t.Fatalf("Error on setup of the test: creating files and folders.")
+	}
+	fmt.Printf("All files in directory")
+	filename, err := printFiles(d)
+	if err != nil {
+		t.Error(err)
+	}
+	fmt.Printf("end of all files")
+	//defer os.RemoveAll(d)
+	fmt.Printf("%v", filename)
+	var flags rmFlags
+	flags.recursive = true
+	if err := rm(filename, flags); err != nil {
 		t.Error(err)
 	}
+	printFiles(d)
+	defer os.RemoveAll(d)
 }
+
+//TEST 4
+/*
+//does not work with Travis, but is still a good test
+func Test_rm_4(t *testing.T) {
```

Comment:
The test works on my local machine but not on travis because it works with a block device I make. Should I still keep it commented or just remove it entirely?